api/rpc/stack/stack-state.go

```diff
@@ -0,0 +1,98 @@
+package stack
+
+import (
+	"fmt"
+	"github.com/appcelerator/amp/api/runtime"
+	"golang.org/x/net/context"
+)
+
+const (
+	// Stopped (initial state)
+	Stopped = iota
+	// Starting state
+	Starting
+	// Running  state
+	Running
+	// Redeploying  state
+	Redeploying
+)
+
+var stackRuleSet = RuleSet{
+	//   | Stopped   | Starting  | Running   | Redeploying
+	[]bool{false /**/, true /* */, false /**/, true /* */}, // Stopped (initial state)
```

Comment:
If you want an array literal, you can let the compiler do the counting: `[...]bool`, but i think a slice literal is fine unless we want to explicitly fix the size.